src/Databases/DataLake/DatabaseDataLake.cpp

```diff
@@ -490,27 +538,40 @@ void registerDatabaseDataLake(DatabaseFactory & factory)
         const auto * database_engine_define = args.create_query.storage;
         const auto & database_engine_name = args.engine_name;
 
+        DatabaseDataLakeSettings database_settings;
+        if (database_engine_define->settings)
+            database_settings.loadFromQuery(*database_engine_define);
+
+        auto catalog_type = database_settings[DB::DatabaseDataLakeSetting::catalog_type].value;
         const ASTFunction * function_define = database_engine_define->engine;
-        if (!function_define->arguments)
+        ASTs engine_args;
+        if (!function_define->arguments && catalog_type != DatabaseDataLakeCatalogType::GLUE)
```

Comment:
Please add a comment that this is because glue catalog does not require endpoint, because it is one per region